lib/Screens/Questionarie/Widgets/generic_quest_room_tile.dart

```diff
@@ -27,51 +27,57 @@ class _GenericQuestRoomTileState extends State<GenericQuestRoomTile> {
           QuestionnaireService()
               .getScoreByWeekApp(widget.email, widget.questCode)
-              .then((scores) => scoreList.addAll(scores))
-              .whenComplete(
-                () => Navigator.of(context).pushNamed(
-                  "/chart-ccsm-screen",
-                  arguments: {
-                    'questName': widget.questName,
-                    'questCode': widget.questCode,
-                    'email': widget.email,
-                    'scoreList': scoreList,
-                  },
-                ),
-              );
+              .then((scores) {
+            scoreList = [];
+            scoreList.addAll(scores);
+          }).whenComplete(
+            () => Navigator.of(context).pushNamed(
+              "/chart-ccsm-screen",
+              arguments: {
+                'questName': widget.questName,
+                'questCode': widget.questCode,
+                'email': widget.email,
+                'scoreList': scoreList,
+              },
+            ),
+          );
         } else if (widget.questCode == QuestionnaireCode.assistn2.name) {
           QuestionnaireService()
               .getScoreByWeekApp(widget.email, widget.questCode)
-              .then((scores) => scoreList.addAll(scores))
-              .whenComplete(
-                () => Navigator.of(context).pushNamed(
-                  "/chart-substance-screen",
-                  arguments: {
-                    'questName': widget.questName,
-                    'questCode': widget.questCode,
-                    'email': widget.email,
-                    'scoreList': scoreList,
-                  },
-                ),
-              );
+              .then((scores) {
+            scoreList = [];
+            scoreList.addAll(scores);
+          }).whenComplete(
+            () => Navigator.of(context).pushNamed(
+              "/chart-substance-screen",
+              arguments: {
+                'questName': widget.questName,
+                'questCode': widget.questCode,
+                'email': widget.email,
+                'scoreList': scoreList,
+              },
+            ),
+          );
         } else if (widget.questCode ==
             QuestionnaireCode.sleepQuestionnaire.name) {
           Navigator.of(context).pushNamed("/chart-sleep-screen");
         } else {
           QuestionnaireService()
               .getScoreByWeekApp(widget.email, widget.questCode)
-              .then((scores) => scoreList.addAll(scores))
-              .whenComplete(
-                () => Navigator.of(context).pushNamed(
-                  "/chart-general-screen",
-                  arguments: {
-                    'questName': widget.questName,
-                    'questCode': widget.questCode,
-                    'email': widget.email,
-                    'scoreList': scoreList,
-                  },
-                ),
-              );
+              .then((scores) {
+            scoreList = [];
```

Comment:
dá pra agrupar todos antes dos if e elses não? logo no inicio do onTap

Reply:
Aqui no caso eu to resetando para ter somente os valores do score, pq tava duplicando